backend.py

```diff
@@ -440,6 +441,19 @@ def performInstallation(answers, ui_package, interactive):
                                "latest dedicated ISO." %
                                (answers['linstor-version'], ', '.join(available_linstor_versions)))
 
+    # if upgrading a host with xapi, check we can upgrade it first
+    if answers['install-type'] == INSTALL_TYPE_REINSTALL and answers['xapi-version']:
+        available_xapi_versions = repository.listPackagesFromRepos(
+            main_repositories, 'xapi-core', '%{evr}', latest_only=True)
+        if not available_xapi_versions:
+            raise RuntimeError("No XAPI package found in package source")
+        assert len(available_xapi_versions) == 1
+        if rpm.labelCompare(('pkg', 'arch', available_xapi_versions[0]), ('pkg', 'arch', answers['xapi-version'])) < 0:
+            raise RuntimeError("Cannot upgrade the host: the XAPI version on the host (%s) "
+                               "is newer than the XAPI version to install (%s). "
+                               "Please refer to the XCP-ng upgrade documentation." %
```

Comment:
Let's not forget to add said doc. We can discuss what we should tell users. I know it's already covered at least for migration from XS, but I'm thinking about users who'd go check the docs after they saw the error. Something that explicitly gives them the error message their search for and explains their options.